core/kernel/user_access.c

```diff
@@ -29,8 +46,11 @@ TEE_Result copy_from_user(void *kaddr, const void *uaddr, size_t len)
 
 	uaddr = memtag_strip_tag_const(uaddr);
 	res = check_access(flags, uaddr, len);
-	if (!res)
+	if (!res) {
+		enter_user_access();
```

Comment:
could this be replaced by a call to `disable_pan()` which is more explicit than `enter_user_access()` regarding what the function is expected to do?

Reply:
This is a non-arch-specific file so PAN should not be used IMHO.

Reply:
It seems to me you said PAN was a generic feature, generic enough for `CFG_PAN` to cover all Privilege Access Never support :)

Reply:
I'd prefer name that are architecture-agnostic, rather than that are specific to ARM architecture. RISC-V also has hardware feature called SSTATUS.SUM (which stands for 'permit Supervisor User Memory access') that is almost equivalent to PAN, and I believe architecture-agnostic naming shall be preferred so that RISC-V can reuse the interface by implementing that function in the future. Some names that I could came up with were `(dis)allow_user_access()` or `(dis)allow_unprivileged_access()`. Do you have more suggestions?

Reply:
I would not use `user_access` because there is no user land access here. It's still core (privileged) access.
Proposition: `allow_core_user_map_accesss()` / `forbid_core_user_map_accesss()`?

Reply:
I guess `enter_user_access()` should be read as "enter a section with user mode memory access temporarily enabled"

Reply:
ok, you have the last word.